code/datums/gamemode/objectives/time_agent_extract.dm

```diff
@@ -0,0 +1,55 @@
+/datum/objective/time_agent_extract
+	name = "extract through anomaly"
+	var/obj/effect/anomaly
+	var/extracted = FALSE
+
+/datum/objective/time_agent_extract/PostAppend()
+	var/list/potential_locations = list()
+	for(var/area/maintenance/A in areas)
+		potential_locations.Add(A)
+	var/placed = FALSE
+	while(!placed && potential_locations.len)
+		var/area/maintenance/A = pick(potential_locations)
+		potential_locations.Remove(A)
+		for(var/turf/simulated/floor/F in A.contents)
+			if(!F.has_dense_content())
+				anomaly = new /obj/effect/time_anomaly(F)
+				placed = TRUE
+				break
+	explanation_text = format_explanation()
+	return TRUE
+
+/datum/objective/time_agent_extract/IsFulfilled()
+	.=..()
+	if(extracted)
+		return TRUE
+
+/datum/objective/time_agent_extract/format_explanation()
+	return "Escape through \the [anomaly], located in [format_text(get_area(anomaly).name)] ([anomaly.x-WORLD_X_OFFSET[anomaly.z]], [anomaly.y-WORLD_Y_OFFSET[anomaly.z]], [anomaly.z]). Use your jump charge to activate it."
+
+/obj/effect/time_anomaly
+	name = "anomaly"
+	desc = "A hole in time and space.<br><span class = 'sinister'>Looking into it is like looking at a picture of yourself looking at a picture of yourself ad infinitum. Looking further, you swear one or two turn their head to look back at you.</span>"
```

Comment:
Sadly I believe desc drops formatting like this so if you want this to appear in sinister text you probably have to move it to examine()